firmware/mcal/stm32f767/periph/pwm.h

```diff
@@ -60,10 +60,22 @@ class PWMOutput : public shared::periph::PWMOutput {
         return float(pulse) / float(period) * 100.0f;
     }
 
+    void SetFrequency(float frequency) override {
+        frequency_ = std::max(0.000015259022f, frequency);
+        uint32_t autoreload = 1/frequency_;
+
+        __HAL_TIM_SetAutoreload(htim, autoreload);
+    }
+
+    float GetFrequency() override {
+        return frequency_;
```

Comment:
This should calculate the frequency from the timer reload and timer speed (see GetTimerFrequency comment above)